AOT+LLVM throws for `Enum.GetUnderlyingType()` on Android in .NET 7

### Description

Our `Mono.Android-Tests` project crashes on startup with:
```
05-05 15:19:34.429  4929  4929 F mono-rt : [ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: Type provided must be an Enum. Arg_ParamName_Name, enumType
05-05 15:19:34.429  4929  4929 F mono-rt :    at System.Enum.InternalGetUnderlyingType(RuntimeType enumType)
05-05 15:19:34.429  4929  4929 F mono-rt :    at System.RuntimeType.GetEnumUnderlyingType()
05-05 15:19:34.429  4929  4929 F mono-rt :    at System.Enum.GetUnderlyingType(Type enumType)
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Interop.JniRuntime.JniTypeManager.GetUnderlyingType(Type , Int32& )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Interop.JniRuntime.JniTypeManager.GetTypeSignature(Type )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Interop.JniPeerMembers.JniInstanceMethods..ctor(Type )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Interop.JniPeerMembers.JniInstanceMethods.GetConstructorsForType(Type )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Interop.JniPeerMembers.JniInstanceMethods.StartCreateInstance(String , Type , JniArgumentValue* )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Lang.Error..ctor(String )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Android.Runtime.JavaProxyThrowable..ctor(Exception )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Java.Lang.Throwable.FromException(Exception )
05-05 15:19:34.429  4929  4929 F mono-rt :    at Android.Runtime.AndroidEnvironment.UnhandledException(Exception e)
05-05 15:19:34.429  4929  4929 F mono-rt :    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V , IntPtr , IntPtr , IntPtr )
```

This looks oddly familiar to a related issue in mono/mono: https://github.com/xamarin/xamarin-android/issues/1842

Looking at the underlying code, I think it throws here: 

https://github.com/xamarin/java.interop/blob/61cdb40dcc8b33d400aef3fd12e0aeee72f71e35/src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs#L151-L152

### Reproduction Steps

Build a .NET 7 `Release` app with `-p:EnableLLVM=true`.

### Expected behavior

Use of `-p:EnableLLVM=true` on Android doesn't crash on startup.

### Actual behavior

Use of `-p:EnableLLVM=true` crashes on startup on Android.

### Regression?

This project appears to be working in .NET 6.

### Known Workarounds

Disable LLVM.

### Configuration

.NET 7.0.100-preview.4.22174.1 c7afae6936bf80239aa93bfd3b6f68513d9876d4
.NET runtime 7.0.0-preview.4.22172.7 c5d40c9e703fd257db1b26ef4fd1399bbae73ab0

### Other information

_No response_

--- a/src/mono/mono/mini/method-to-ir.c
+++ b/src/mono/mono/mini/method-to-ir.c
@@ -2232,7 +2232,16 @@ static MonoInst*
 mono_emit_widen_call_res (MonoCompile *cfg, MonoInst *ins, MonoMethodSignature *fsig)
 {
 	if (!MONO_TYPE_IS_VOID (fsig->ret)) {
-		if ((fsig->pinvoke || LLVM_ENABLED) && !m_type_is_byref (fsig->ret)) {
+		// FIXME
+		// LLVM code doesn't uses zero extend the full word while jit expects it.
+		// A proper fix would be to detect if we are actually using llvm code from aot images
+		// or make sure llvm code actually zero extends the return.
+#ifdef MONO_ARCH_LLVM_SUPPORTED
+		gboolean might_use_llvm = TRUE;
+#else
+		gboolean might_use_llvm = FALSE;
+#endif
+		if ((fsig->pinvoke || might_use_llvm) && !m_type_is_byref (fsig->ret)) {
 			int widen_op = -1;
 
 			/*

